tests/test_classification.py

```diff
@@ -779,3 +857,620 @@ def test_sampling_for_multinomialNB():
+    # same other fixed parameters that we used for SKLL
+    models_with_C_values = {}
+    for param_value in grid_search_results['params']:
+        model_kwargs = param_value
+        model_kwargs.update({'max_iter': 1000,
+                             'solver': 'liblinear',
+                             'multi_class': 'auto',
+                             'random_state': 123456789})
+        sklearn_learner = LogisticRegression(**model_kwargs)
+        models_with_C_values[param_value['C']] = sklearn_learner
+
+    # now let's split the featureset the same way SKLL would have
+    # done using the folds file
+    dummy_label = next(iter(folds_dict.values()))
+    fold_groups = [folds_dict.get(curr_id, dummy_label) for curr_id in shuffled_fs.ids]
+    kfold = FilteredLeaveOneGroupOut(folds_dict, shuffled_fs.ids)
+    fold_train_test_ids = list(kfold.split(shuffled_fs.features, shuffled_fs.labels, fold_groups))
+
+    # generate predictions on the test split of the each fold, and
+    # then compute the objectives appropriately (using either the
+    # labels or the probabilities) and store them for comparison
+    metric_values_dict = {}
+
+    # compute values for each fold
+    for fold_id in [0, 1]:
+
+        metric_values_dict[fold_id] = []
+
+        fold_train_ids = fold_train_test_ids[fold_id][0]
+        fold_test_ids = fold_train_test_ids[fold_id][1]
+
+        X_fold_train = X[fold_train_ids, :]
+        X_fold_test = X[fold_test_ids, :]
+
+        y_fold_train = y[fold_train_ids, ]
+        y_fold_test = y[fold_test_ids, ]
+
+        # let's also compute the class/label indices
+        # since we will need them later
+        y_fold_train_indices = [clf.label_dict[label] for label in y_fold_train]
+        y_fold_test_indices = [clf.label_dict[label] for label in y_fold_test]
+
+        # iterate over the two trained sklearn models;
+        # one for each point on the C grid
+        for C_value in models_with_C_values:
+
+            # for training, we use the class indices
+            # rather than the class labels like SKLL does
+            # so that our predictions are in the index
+            # space rather than label space
+            sklearn_learner = models_with_C_values[C_value]
+            sklearn_learner.fit(X_fold_train, y_fold_train_indices)
+
+            # compute both labels and probabilities via sklearn
+            sklearn_fold_test_labels = sklearn_learner.predict(X_fold_test)
+            sklearn_fold_test_probs = sklearn_learner.predict_proba(X_fold_test)
+
+            # now let's proceed on a metric by metric basis
+            # and note that we are again using the class
+            # indices rather than the class labels themselves
+
+            # 1. Correlation metrics are only computed for integer
+            #    or float labels; they use probability values if
+            #    available only in the binary case.
+            if metric_name == 'pearson':
```

Comment:
Nitpick: There is some shared code between the blocks related to the conditions for Pearson, Spearman, and Kendall-Tau. They could probably be collapsed and a `metric_func`-type variable could be used to loop over the 3 possible metrics.

Reply:
@jbiggsets had the same suggestion! I just pushed a commit that does this. Thanks!